DataProcessor/DataUnpacker.cpp

```diff
@@ -92,9 +92,12 @@ DataUnpacker::DataUnpacker(QObject *parent) : QObject(parent)
     connect(&dataHandlingThread, &QThread::started, retriever, &BackendProcesses::startThread);
     connect(this, &DataUnpacker::getData, retriever, &BackendProcesses::threadProcedure);
     connect(retriever, &BackendProcesses::dataReady, this, &DataUnpacker::unpack);
+    connect(retriever, &BackendProcesses::eng_dash_connected, this, &DataUnpacker::eng_dash_connected);
+    connect(retriever, &BackendProcesses::eng_dash_disconnected, this, &DataUnpacker::eng_dash_disconnected);
```

Comment:
Consider using one slot and signal pair with boolean arguments to indicate the connection status rather than two separate pairs of slots and signals. For example, you could use a signal/slot pair like eng_dash_connection_status_changed(bool connected) and pass the connection status as an argument when emitting the signal and receiving it in the slot. Then, you could just set eng_dash_commfail equal to the value passed (or its complement).